src/rules/valid-scripts.ts

```diff
@@ -0,0 +1,69 @@
+import type { AST as JsonAST } from "jsonc-eslint-parser";
+
+import * as ESTree from "estree";
+
+import { createRule } from "../createRule.js";
+
+export const rule = createRule({
+	create(context) {
+		return {
+			"Program > JSONExpressionStatement > JSONObjectExpression > JSONProperty[key.value=scripts]"(
+				node: JsonAST.JSONProperty,
+			) {
+				const { value } = node;
+				if (value.type !== "JSONObjectExpression") {
+					context.report({
+						messageId: "incorrectTypeScriptsField",
+						node: node.value as unknown as ESTree.Node,
+					});
+					return;
+				}
+				if (value.properties.length === 0) {
+					context.report({
+						messageId: "emptyScriptsField",
+						node: node.value as unknown as ESTree.Node,
+					});
+					return;
+				}
+				for (const prop of value.properties) {
+					const { value } = prop; // key,
+					if (
+						value.type !== "JSONLiteral" ||
+						typeof value.value !== "string"
+					) {
+						context.report({
+							messageId: "incorrectTypeScript",
+							node: node.value as unknown as ESTree.Node,
+						});
+						return;
```

Comment:
[Question] Why `return;`? If multiple scripts are incorrect, we'd want to get reports on all of them I think.

E.g. here I'd expect the values of both `script1` and `script3` to have complaints:

```json
"scripts": {
    "script1": "",
    "script2": "...",
    "script3": "",
}
```

(here and the other `return`)